packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m

```diff
@@ -133,15 +137,8 @@ - (void)signIn:(GIDSignIn *)signIn
     didSignInForUser:(GIDGoogleUser *)user
            withError:(NSError *)error {
   if (error != nil) {
-    if (error.code == kGIDSignInErrorCodeHasNoAuthInKeychain ||
-        error.code == kGIDSignInErrorCodeCanceled) {
-      // Occurs when silent sign-in is not possible or user has cancelled sign
-      // in,
-      // return an empty user in this case
-      [self respondWithAccount:nil error:nil];
-    } else {
-      [self respondWithAccount:nil error:error];
-    }
+    // Forward all errors and let Dart side decide how to handle.
+    [self respondWithAccount:nil error:error];
```

Comment:
Doesn't look like it.

On Android 

- `CommonStatusCodes.SIGN_IN_REQUIRED = 4`
- `GoogleSignInStatusCodes.SIGN_IN_CANCELLED = 12501`

On iOS:

- `kGIDSignInErrorCodeHasNoAuthInKeychain = -4`
- `kGIDSignInErrorCodeCanceled = -5`

We could probably normalize these error codes to be consistent across platforms as they seem to have the same meaning. What would be the right way to do this?

Reply:
Reporting an error back to Dart using iOS `result(someFlutterError)` and Android `result.error(x, y, z)` has the effect that `await invokeMethod(...);` throws a `PlatformException` on the Dart side. This may be just what we want. Doing it consistently then involves ensuring that the three fields of the error (code, message, details) are the same across iOS and Android, or at least that the error code part is. Indeed, one might just define a suitable set of error code strings to populate the first error component and leave the other two fields null.

For some cases (like the user cancelling the sign in dialog?), it may not be desirable to throw an exception on the Dart side. If so, one must instead return a successful result that is distinguishable from a user account, e.g. null, using `result(nil)` on iOS and `result.success(null)` on Android.

Your call.

Reply:
I like it, and I think we should extend the approach to the "catch-all" error code. There is no established convention yet, and each plugin can do what it pleases. But I do think that consolidating on the `PlatformException` error codes between Android and iOS for a single plugin will make that plugin much easier to use for client Dart code.

Reply:
I consolidated "catch-all" case as `sign_in_failed` and pushed an update.

However I realized that there is a bit of a breaking change in `signIn` method as previously it was silently ignoring `sign_in_canceled` scenario. I updated it with logic similar to `signInSilently`, but it is still not exactly the same at it was before.

Previously it'd fail on errors other than `sign_in_canceled`. Now it would return `null` for all errors by default, while users can opt-in to handle all errors with `suppressErrors` set to `false`.

Reply:
I suppose we could restore that behavior by not treating `sign_in_cancelled` as an error, but instead return a `null` user account as a success result?